Cargo.toml

```diff
@@ -48,6 +50,7 @@ lto = 'off'
 
 [profile.release]
 debug = true
+rustflags = ["-Ctarget-cpu=native"] #5367
 
```

Comment:
So, according to the priority list in the PR description, am I right to say that this is considered build.rustflags and hence has lower priority than target.rustflags?

Reply:
This is a new feature named "profile rustflags" which is not mentioned in the doc. It's joined with 3).

```
     Running `rustc --crate-name regex_syntax --edition=2018 /home/bugenzhao/.cargo/registry/src/github.com-1ecc6299db9ec823/regex-syntax-0.6.27/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -Ctarget-cpu=native -C embed-bitcode=no -C debuginfo=2 -C debug-assertions=off --cfg 'feature="default"' --cfg 'feature="unicode"' --cfg 'feature="unicode-age"' --cfg 'feature="unicode-bool"' --cfg 'feature="unicode-case"' --cfg 'feature="unicode-gencat"' --cfg 'feature="unicode-perl"' --cfg 'feature="unicode-script"' --cfg 'feature="unicode-segment"' -C metadata=9940732c2dd3c8e5 -C extra-filename=-9940732c2dd3c8e5 --out-dir /home/bugenzhao/work/risingwave/target/release/deps -L dependency=/home/bugenzhao/work/risingwave/target/release/deps --cap-lints allow -Ctarget-feature=+avx2 --cfg tokio_unstable -Funused_must_use '-Wclippy::dbg_macro' '-Wclippy::disallowed_methods' '-Wclippy::disallowed_types' '-Wclippy::doc_markdown' '-Wclippy::explicit_into_iter_loop' '-Wclippy::explicit_iter_loop' '-Wclippy::inconsistent_struct_constructor' '-Wclippy::unused_async' '-Wclippy::map_flatten' '-Wclippy::no_effect_underscore_binding' '-Wclippy::await_holding_lock' '-Wrustdoc::broken_intra_doc_links' -Wfuture_incompatible -Wnonstandard_style -Wrust_2018_idioms -Clink-arg=-fuse-ld=lld -Clink-arg=-Wl,--no-rosegment`
```

Reply:
Oh I see. In this case target-cpu native may not be desirable. As it would mean we may have a cross-target compatible binary, the compatibility of which we would want in a release binary.

We probably only want this flag if the target does not correspond to aarch or x86. I'm not sure if target triple `target.x86_64-unknown-linux-gnu` also includes x86 on mac and windows?

Reply:
Okay. Let's remove this flag. I believe we only care about the performance under Linux with x86-64 and aarch64 for now. Let's tweak it in the future, if necessary.